lib/vagrant-libvirt/action/remove_stale_volume.rb

```diff
@@ -20,20 +20,20 @@ def call(env)
 
           config = env[:machine].provider_config
           # Check for storage pool, where box image should be created
-          fog_pool = ProviderLibvirt::Util::Collection.find_matching(
-            env[:machine].provider.driver.connection.pools.all, config.storage_pool_name
-          )
-          @logger.debug("**** Pool #{fog_pool.name}")
+          fog_pool = env[:machine].provider.driver.connection.pools.all(
+            name: config.storage_pool_name
+          ).first
+          @logger.debug("**** Pool #{fog_pool.nil? ? 'not found' : fog_pool.name}")
```

Comment:
Suggested to someone in another review, it might be easier if this just calls `@app.call(env)` with `env[:result] = nil` if fog_pool is nil at this point. Return early rather than continuing to process.

Reply:
though worth printing out the debug message first anyway, so just an early call to @app.call(env) here, might see if it can be added via a different PR and you can then fix this one up

Reply:
let me know if my solution is too longwinded and I can remove it and wait for the other PR

Reply:
I updated to rebase on top of another merged version, think I got the same logic in place